stable/oauth2-proxy/templates/deployment.yaml

```
@@ -20,6 +20,7 @@ spec:
         checksum/config-emails: {{ include (print $.Template.BasePath "/configmap-authenticated-emails-file.yaml") . | sha256sum }}
         checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
         checksum/google-secret: {{ include (print $.Template.BasePath "/google-secret.yaml") . | sha256sum }}
+        checksum/htpasswd: {{ include (print $.Template.BasePath "/configmap-htpasswd-file.yaml") . | sha256sum }}
```

```suggestion
{{- if .Values.htpasswdFile.enabled }}
        checksum/htpasswd: {{ include (print $.Template.BasePath "/configmap-htpasswd-file.yaml") . | sha256sum }}
{{- end }}
```

We do not want this if htpasswd is disabled

---

I did this too initially, but then removed it because even if the `if` condition is not there, it will just hash the default `configmap-htpasswd-file.yaml`, which will always be constant unless modified (which will happen only if `.Values.htpasswdFile.enabled` is `true`).

Anyways, either way is fine with me, it works both the ways.

---

IMHO, Although it technically works, it would be confusing for users who may wonder if it is really enabled or not.

---

Yes, it increases the readability for sure. Added your changes.